### PR TITLE
First stab at handling console messages while a test runs

### PR DESCRIPTION
**Pull request by DmitrySharabin**
In this PR, I tried to implement [@LeaVerou's suggestion on handling console messages from tests](https://github.com/LeaVerou/htest/issues/7#issuecomment-2141051134), including (as an addition to the main algorithm):

- Refactor `interceptConsole()` so that it accepts a function as an argument
- Add utility `pluralize()` method to correctly print out “message”/“messages” to the console

Here is what we have as a starting point:

https://github.com/LeaVerou/htest/assets/9166277/61478182-77c6-4832-a0f8-8924d09deef9

Let's improve it and make it more awesome together!

### CODE REVIEW

File: src/env/node.js [hunk 1]
```diff
@@ -10,20 +10,25 @@ import { globSync } from 'glob';
...
 
-// Recursively traverse a subtree starting from `node` and make (only) groups of tests collapsible
+/**
+ * Recursively traverse a subtree starting from `node` and make groups of tests and test with console messages collapsible.
+ */
 function makeCollapsible (node) {
-	if (node.tests?.length) {
-		node.collapsed = true; // all groups are collapsed by default
+	if (node.tests?.length || node.messages?.length) {
+		node.collapsed = true; // all groups and test with console messages are collapsed by default
```
Comment (LeaVerou) [inline src/env/node.js:20]:
```suggestion
		node.collapsed = true; // all groups and console messages are collapsed by default
```

**Review by LeaVerou — CHANGES_REQUESTED** [review]
Code seems ok ("console message" is a bit of an odd term, but not sure what else we could use. "Console output" is more fitting but not countable, so we can't say 3 console outputs.). Not a fan of the multiple sequential parens. I don't think the console messages need parens, a separator is probably fine, or even a comma.

**Comment by DmitrySharabin** [comment]
> "console message" is a bit of an odd term, but not sure what else we could use. "Console output" is more fitting but not countable, so we can't say 3 console outputs.

How about “N output messages,” “N log messages,” or simply “N messages,” then?

> Not a fan of the multiple sequential parens.

Me neither.

What if we move info about console messages _before_ the time spent? It'll become aligned with the summary we provide for groups.

Like so:

image

**Comment by LeaVerou** [comment]
> > "console message" is a bit of an odd term, but not sure what else we could use. "Console output" is more fitting but not countable, so we can't say 3 console outputs.
> 
> How about “N output messages,” “N log messages,” or simply “N messages,” then?

Not a fan of "log messages", since they may not all be log messages. Anyhow, that’s not a blocker, we can use whatever for now and change it very easily.

> > Not a fan of the multiple sequential parens.
> 
> Me neither.
> 
> What if we move info about console messages _before_ the time spent? It'll become aligned with the summary we provide for groups.
> 
> Like so:
> 
> image

Much better!

**Comment by DmitrySharabin** [comment]
> Not a fan of "log messages", since they may not all be log messages. Anyhow, that’s not a blocker, we can use whatever for now and change it very easily.

I went with “messages.” Yeah, we can change it anytime.

It's ready for your other review, then.

**Review by LeaVerou — APPROVED** [review]

**Pull request merged**